### PR TITLE
Update README.md

### LINKED ISSUE #1: Add readme

### PR DESCRIPTION
**Pull request by IttayEyal**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,42 @@
 Memoization decorator for python
```
Comment (adamatan) [inline README.md:1]: Consider "Persistent Python memoization". The decorator is merely the technique.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,42 @@
 Memoization decorator for python 
-Store function results on disk. Load if call to the function was previously stored. 
+=== 
+Add @Memoized before function definition for automatic caching.
```
Comment (adamatan) [inline README.md:3]: 1. `@Memoized`, not  @Memoized 
2. The first thing should be a dead-simple example IMO, not explanations.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,42 @@
 Memoization decorator for python 
-Store function results on disk. Load if call to the function was previously stored. 
+=== 
+Add @Memoized before function definition for automatic caching. 
+
+On the first call to the function the decorator runs it with the provided parameters. Then it stores the returned result in a file. In subsequent calls to the function with the same parameters, the decorator returns the results from the cache file without calling the function. The decorator matches the function text and the parameters, so if the function or the parameters change the function will be run again.
```
Comment (adamatan) [inline README.md]: function text -> function name
Consider "The file name, function name, and the function arguments."

File: README.md [hunk 4]
```diff
@@ -1,2 +1,42 @@
...
-Store function results on disk. Load if call to the function was previously stored. 
+=== 
+Add @Memoized before function definition for automatic caching. 
+
+On the first call to the function the decorator runs it with the provided parameters. Then it stores the returned result in a file. In subsequent calls to the function with the same parameters, the decorator returns the results from the cache file without calling the function. The decorator matches the function text and the parameters, so if the function or the parameters change the function will be run again. 
+
+    from memoize.memoized import Memoized
+    import time
+
+    @Memoized(cacheDir = "/home/username/tmp") # cacheDir defaults to /tmp in Linux
```
Comment (adamatan) [inline README.md:10]: Why complicate the first example with an argument that will be rarely used?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,42 @@
...
+Add @Memoized before function definition for automatic caching. 
+
+On the first call to the function the decorator runs it with the provided parameters. Then it stores the returned result in a file. In subsequent calls to the function with the same parameters, the decorator returns the results from the cache file without calling the function. The decorator matches the function text and the parameters, so if the function or the parameters change the function will be run again. 
+
+    from memoize.memoized import Memoized
+    import time
+
+    @Memoized(cacheDir = "/home/username/tmp") # cacheDir defaults to /tmp in Linux 
+    def fibonacci(n):
+        if n <= 2: return 1
```
Comment (adamatan) [inline README.md:12]: Let's use a shorter, idiomatic implementation - see [here](https://stackoverflow.com/a/15047141/51197). For example, we can do `a, b, k = 1, 1, 2`.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,42 @@
...
+    n = 3e5
+    start = time.time()
+    runOne = fibonacci(n)
+    finish = time.time()
+    print("Runtime 1: %.2f" % (finish-start))
+
+    start = time.time()
+    runTwo = fibonacci(n)
+    finish = time.time()
+    print("Runtime 2: %.2f" % (finish-start))
```
Comment (adamatan) [inline README.md:32]: Or, even more simple - run the two cases in a loop:

```python
for I in range(5):
    start = time.time() 
    result = fibonacci(n)
    finish = time.time()
    print(f'Runtime {i}: f'{finish-start:.2f}')
```

```
    Runtime 0: 0.74
    Runtime 1: 0.00
    Runtime 1: 0.00
    Runtime 1: 0.00
    Runtime 1: 0.00   
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,42 @@
...
+
+    assert runOne == runTwo
```
Comment (adamatan) [inline README.md:34]: Just print it in the loop. More intuitive and more convincing.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,42 @@
...
+
+Output: 
+
+    Runtime 1: 0.74
+    Runtime 2: 0.00
+
+Set `Memoized.readCache` to False to skip reading the cache (but store results there nonetheless).
```
Comment (adamatan) [inline README.md:41]: Does not belong in the first example IMO, and we should rethink this parameter.

**Review by adamatan — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (IttayEyal) [reply README.md]: The function code, actually

File: README.md (hunk 1 above)
Comment (IttayEyal) [reply README.md:1]: Clarify it's a decorator from the get go.

File: README.md (hunk 2 above)
Comment (IttayEyal) [reply README.md:3]: 1. I think you fixed the capitalization already. 
2. Looks good.

File: README.md (hunk 4 above)
Comment (IttayEyal) [reply README.md:10]: Just one example...

File: README.md (hunk 5 above)
Comment (IttayEyal) [reply README.md:12]: done.

File: README.md (hunk 6 above)
Comment (IttayEyal) [reply README.md:32]: That's how it started. I like the linear version better.

File: README.md (hunk 7 above)
Comment (IttayEyal) [reply README.md:34]: No loop.

File: README.md (hunk 8 above)
Comment (IttayEyal) [reply README.md:41]: Only one example.

**Pull request merged**